### PR TITLE
Assert on decoded WebSocket messages instead of literal JSON

### LINKED ISSUE #18: Improve assertions on received messages in websocket_test.exs
Improve the way we make assertions on received messages (e.g. `assert_receive` wants pattern match and not functions or variables) in the `websocket_test.exs`.

### PR DESCRIPTION
**Pull request by joebew42**
## Context

The WebSocket acceptance tests assert on messages received from the server as hard-coded literal JSON strings passed to `assert_receive`. Because `assert_receive/3` expects a pattern rather than an arbitrary expression, the expected payload has to be inlined as an exact escaped string. This makes the assertions brittle — they depend on the precise JSON key ordering produced by the serializer, so an encoding change breaks a test even when the semantic content is unchanged — as well as noisy to read and impossible to compose from variables.

## This PR

This PR asserts on the decoded content of each received message rather than on its exact serialized form. A small `assert_receive_message` helper receives the next forwarded text frame, decodes it, and compares the resulting map to the expected map, so assertions are independent of JSON key ordering and read as a single expressive line. The helper matches selectively — non-matching frames are left in the mailbox — and a companion `refute_receive_message` covers the negative case, keeping the existing tests for interleaved broadcasts and duplicate joins working.

Closes #18

### CODE REVIEW

File: test/ex_chat/web/websocket_acceptance_test.exs [hunk 1]
```diff
@@ -182,10 +184,66 @@ defmodule ExChat.Web.WebSocketAcceptanceTest do
       {:ok, client} = connect_to websocket_chat_url(with: body["access_token"]), forward_to: self()
       send_as_text(client, "{\"command\":\"join\"}")
 
-      assert_receive "{\"room\":\"default\",\"message\":\"welcome to the default chat room, charlie!\"}"
+      assert_receive_message %{"room" => "default", "message" => "welcome to the default chat room, charlie!"}
+    end
+  end
+
+  # Asserts a forwarded text frame whose decoded content equals `expected` is
```
Comment (joebew42) [inline test/ex_chat/web/websocket_acceptance_test.exs]: Maybe we can remove these comments from here. Not strictly needed.

File: test/ex_chat/web/websocket_acceptance_test.exs [hunk 2]
```diff
@@ -182,10 +184,66 @@ defmodule ExChat.Web.WebSocketAcceptanceTest do
...
+        requeue(Enum.reverse(skipped))
+        decoded
+
+      {:timeout, skipped} ->
+        requeue(Enum.reverse(skipped))
+        flunk("expected to receive a message matching #{inspect(expected)}")
+    end
+  end
+
+  # Refutes receiving a text frame whose decoded content equals `unexpected`,
```
Comment (joebew42) [inline test/ex_chat/web/websocket_acceptance_test.exs]: Same here, I would suggest to remove the comments.

**Review by joebew42 — COMMENTED** [review]
There are a few comments in code that I would suggest removing.

File: test/ex_chat/web/websocket_acceptance_test.exs (hunk 1 above)
Comment (joebew42) [reply test/ex_chat/web/websocket_acceptance_test.exs]: Done — removed the explanatory comment block above `assert_receive_message` in commit a3b3937. The helper name and its small body are self-explanatory, so the comment wasn't strictly needed.

File: test/ex_chat/web/websocket_acceptance_test.exs (hunk 2 above)
Comment (joebew42) [reply test/ex_chat/web/websocket_acceptance_test.exs]: Done — removed this comment block above `refute_receive_message` as well in commit a3b3937, for the same reason. All 45 tests still pass.
